### PR TITLE
IP-283: Scrubbing the routemaster pending queue

### PR DESCRIPTION
**Pull request by mezis**
If a worker is ever SIGKILL'd, it'll potentially leave behind an unprocessed job. This diff:

- regularly scrubs the pending jobs for dead workers,
- clears dead worker metadata, and
- adds monitoring for both

Follow-up to #50 and branched off it (easier to read the [partial diff](https://github.com/deliveroo/routemaster/compare/IP-425/more-metrics...deliveroo:IP-283/scrubbing-the-routemaster-pending-queue)).

===

Jira story [#IP-283](https://deliveroo.atlassian.net/browse/IP-283) in project *Infrastructure and Platform*:

> ## What
> 
> If the watch process dies uncleanly for any reason, it's possibly that un-acked messages will still be in the "pending" queues.
> 
> Add a process that regularly scrubs "pending" queues (ie. auto-nacks the messages).
> 
> ## How
> 
> Given messages can stay "pending" for the subscription batching timeout (t1) + the event delivery HTTP timeout (t2, not yet enforced), it's reasonable to scrub any pending messages that are older than 4 ×(t1 + t2).
> 
> (Ref: https://github.com/deliveroo/routemaster/issues/8)

### CODE REVIEW

File: routemaster/jobs/fail.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    class Fail
+      include Mixins::Log
+
+      def call(*args)
+        raise 'failing job'
+      end
+    end
+  end
+end
+
```
Comment (FriedSock) [inline routemaster/jobs/fail.rb]: What is this job for? I don't see where it is being used.

edit: Ah, I see it is just used in a test case, is it possible to just define this class inside the spec?

Comment (mezis) [reply routemaster/jobs/fail.rb]: Not easily. Job loading is conventional - it loads the job class using `require "routemaster/jobs/#{name}"`.
Any alternate suggestions?

Comment (FriedSock) [reply routemaster/jobs/fail.rb]: I can't see a nice way unless we manually require all the jobs in that directory at the top of the file for the `Job` class, what is the reason for not doing that, startup time?

Comment (mezis) [reply routemaster/jobs/fail.rb]: No good reason as the job list is short. And startup time wouldn't be changed much!
Thanks for prodding, I'll tweak this.

**Review by FriedSock — APPROVED** [review]

**Pull request merged**